junit-platform-console/src/main/java/org/junit/platform/console/tasks/StdStreamHandler.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2015-2025 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.console.tasks;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import org.junit.platform.commons.JUnitException;
+
+class StdStreamHandler implements AutoCloseable {
+	private PrintStream stdout;
+	private PrintStream stderr;
+
+	public StdStreamHandler() {
+	}
+
+	private boolean isSameFile(Path path1, Path path2) {
+		if (path1 == null || path2 == null)
+			return false;
+		return (path1.normalize().toAbsolutePath().equals(path2.normalize().toAbsolutePath()));
+	}
+
+	public void redirectStdStreams(Path stdoutPath, Path stderrPath) {
+		if (isSameFile(stdoutPath, stderrPath)) {
+			try {
+				PrintStream commonStream = new PrintStream(Files.newOutputStream(stdoutPath), true);
```

Comment:
We should document that the default charset is used to write these files. We might later introduce an option to configure the charset, if necessary, but I don't think we need to do so right away.